test/data/jobsData.js

```diff
@@ -363,11 +369,161 @@ define(['common/format'], (format) => {
         return acc;
     }, {});
 
+    /**
+     * Increment all timestamps in an object by ${time} seconds
+     *
+     * @param Object jobState
+     * @param int time -- how much to increment any timestamps by
+     */
+    function passTime(jobState, time) {
+        ['created', 'finished', 'queued', 'running', 'updated'].forEach((timePoint) => {
+            if (jobState[timePoint]) {
+                jobState[timePoint] += time;
+            }
+        });
+    }
+
+    // convert a job into one that can be (has been) retried
+    function convertToRetryParent(jobState, retryArray = []) {
+        jobState.meta.originalJob = true;
+        addRetries(jobState, retryArray);
+    }
+
+    // convert a job into one that is a retry of ${retryParentId}
+    function convertToRetry(jobState, retryParentId) {
+        jobState.meta.retryTarget = retryParentId;
+        jobState.retry_parent = retryParentId;
+        addRetries(jobState, []);
+    }
+
+    function addRetries(jobState, retryArray = []) {
+        jobState.retry_ids = retryArray;
+        jobState.retry_count = retryArray.length;
+    }
+    /**
+     * output of createBatchJob:
+     *
+     * batch parent: 'job-created'
+     *
+     * initial children:
+     * 'job-cancelled-whilst-in-the-queue'
+     * 'job-cancelled-during-run'
+     * 'job-died-whilst-queueing'
+     * 'job-in-the-queue'
+     *
+     * job retries:
+     * 'job-cancelled-whilst-in-the-queue'
+     *  - retry 1: 'job-running'
+     *
+     * 'job-cancelled-during-run'
+     *  - retry 1: 'job-finished-with-success'
+     *
+     * 'job-died-whilst-queueing'
+     *  - retry 1: 'job-died-with-error'
+     *  - retry 2: 'job-estimating' (most recent retry)
+     *
+     * Extra metadata for batch jobs:
+     * meta.currentJob: true/false -- this is the most recent job (including retries)
+     * meta.originalJob: true/false -- this is one of the original jobs in the batch
+     */
+
+    function createBatchJob() {
+        // extra metadata:
+        //
```

Comment:
I feel like there's some added suspense here! (or maybe that line isn't needed, as the info is above).

Reply:
hah! Those invisible characters catch people out every time!